src/main/java/mate/academy/springboot/datajpa/service/impl/ProductServiceImpl.java

```
@@ -0,0 +1,67 @@
+package mate.academy.springboot.datajpa.service.impl;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Map;
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.repository.ProductRepository;
+import mate.academy.springboot.datajpa.service.ProductService;
+import mate.academy.springboot.datajpa.specification.SpecificationManager;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ProductServiceImpl implements ProductService {
+    private final SpecificationManager<Product> productSpecificationManager;
+    private final ProductRepository productRepository;
+
+    public ProductServiceImpl(SpecificationManager<Product> productSpecificationManager,
+                              ProductRepository productRepository) {
+        this.productSpecificationManager = productSpecificationManager;
+        this.productRepository = productRepository;
+    }
+
+    @Override
+    public Product save(Product product) {
+        return productRepository.save(product);
+    }
+
+    @Override
+    public List<Product> getAll() {
+        return productRepository.findAll();
+    }
+
+    @Override
+    public List<Product> findAllByPriceBetween(BigDecimal from, BigDecimal to) {
+        return productRepository.findAllByPriceBetween(from, to);
+    }
+
+    @Override
+    public Product get(Long id) {
+        return productRepository.getById(id);
```

let's throw an exception in case product isn't found